packages/cadmium/src/extrusion.rs

```diff
@@ -40,12 +36,16 @@ use crate::sketch::Vector2;
 
 // use truck_meshalgo::prelude::*;
 use truck_modeling::{
-    builder, Edge, Face as TruckFace, Point3 as TruckPoint3, Vector3 as TruckVector3, Vertex, Wire,
+    builder, Edge, Face as TruckFace, Point3 as TruckPoint3, Vector3 as TruckVector3,
+    Vertex, Wire, Curve, Surface, Plane, builder::translated
 };
 
+use truck_topology::Solid as TruckSolid;
+
 const MESH_TOLERANCE: f64 = 0.1;
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
+#[derive(Tsify, Debug, Clone, Serialize, Deserialize)]
+#[tsify(into_wasm_abi, from_wasm_abi)]
```

Comment:
I figured rust is typed and I would want a tool to generate types for me so therefore some blessed soul will have already written something! I looked at a few options and this one seemed easy - another one seemed too complicated because you first convert rust types to json schema and then with another tool convert to typescript. I imagine since it's json schema you can easily create runtime type guards automatically too. It's probably a good idea to do something like that later - generate all types and typechecking functions for typescript at the wasm rust end, but for now these two simple tools plus some copy paste are all good :)